tracee/tracee.bpf.c

```diff
@@ -627,15 +647,23 @@ static __always_inline int save_context_to_buf(buf_t *submit_p, void *ptr)
     return 0;
 }
 
-static __always_inline context_t init_and_save_context(buf_t *submit_p, u32 id, u8 argnum, long ret)
+static __always_inline context_t init_and_save_context(void* ctx, buf_t *submit_p, u32 id, u8 argnum, long ret)
 {
     context_t context = {};
     init_context(&context);
     context.eventid = id;
     context.argnum = argnum;
     context.retval = ret;
-    save_context_to_buf(submit_p, (void*)&context);
 
+    // Get Stack trace
+    if (get_config(CONFIG_CAPTURE_STACK_TRACES)) {
+        u32 stack_id = bpf_get_stackid(ctx, &stack_traces, BPF_F_REUSE_STACKID | BPF_F_USER_STACK);
+        if (stack_id >= 0) {
```

Comment:
Out of curiousity, when does stack_id<0?

Reply:
Sorry, this should be a signed int. From [doco](https://www.man7.org/linux/man-pages/man7/bpf-helpers.7.html):
```
The stack id retrieved is a 32 bit long integer handle
...
Return: The positive or null stack id on success, or a negative error in case of failure.
```

Reply:
I made the decision not to care about errors on the eBPF side, to ensure we don't impact performance as much